tests/components/mqtt/test_init.py

```diff
@@ -728,6 +745,27 @@ def record_calls(topic, payload, qos):
 
     await hass.async_block_till_done()
     assert len(calls) == 1
+    mqtt_mock.async_publish.reset_mock()
+
+    # Test with partial wrapper
+    calls = []
+    unsub = await mqtt.async_subscribe(
+        hass, "test-topic", RecordCallsPartial(record_calls)
```

Comment:
Maybe pass in a coroutine function instead of a callback function so we test that coroutine or not is detected correctly?

Reply:
Changed signature:
```python
async def record_calls(topic, payload, qos):
  ...
```

Reply:
Somehow this was already correct. Can you have another look now. All code is tested now.